coderd/workspaces.go

```diff
@@ -883,18 +883,9 @@ func validWorkspaceDeadline(old, new time.Time) error {
 		return xerrors.New("nothing to do: no existing deadline set")
 	}
 
-	now := time.Now()
-	if new.Before(now) {
-		return xerrors.New("new deadline must be in the future")
-	}
-
-	delta := new.Sub(old)
-	if delta < time.Minute {
-		return xerrors.New("minimum extension is one minute")
-	}
-
-	if delta > 24*time.Hour {
-		return xerrors.New("maximum extension is 24 hours")
```

Comment:
*review:* Dropping these validations for the moment till we have more information from  https://github.com/coder/coder/issues/2318

Reply:
Actually, there's one very important validation involving templates that I may have overlooked! Adding this in now.